libutils/mutex.c

```diff
@@ -51,30 +51,26 @@ int __ThreadLock(pthread_mutex_t *mutex,
 
     if (result != 0)
     {
-        /* Log() is blocking on mutexes itself inside malloc(), so maybe not
-         * the best idea here. */
-        Log(LOG_LEVEL_ERR,
+        fprintf(stderr,
             "Locking failure at %s:%d function %s! (pthread_mutex_lock: %s)",
             filename, lineno, funcname, GetErrorStrFromCode(result));
-        return false;
+        exit(255);
```

Comment:
This is probably not a good idea.

Reply:
I think that's good. Considering the situations you mentioned, I can't think of a better resolution. I would prefer exit code 101 though. This is what cf-hub uses for unexpected / programming error.